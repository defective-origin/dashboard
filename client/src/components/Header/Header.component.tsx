import React from 'react'

// ---| core |---
import { cn } from 'tools'

// ---| pages |---
// ---| screens |---
// ---| components |---
import Block, { BlockProps } from 'components/Block'

// ---| self |---
import css from './Header.module.scss'

export type HeaderProps = BlockProps

/**
 * Component description.
 *
 * How to use
 * @example
 * <Header />
 */
export function Header(props: HeaderProps): JSX.Element {
  const { children, className, ...otherProps } = props
  const _className = cn(css.Header, className)

  return <Block className={_className} area='top' {...otherProps}>{children}</Block>
}

Header.displayName = 'Header'

export default Header
