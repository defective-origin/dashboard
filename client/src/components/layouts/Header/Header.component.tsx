import React from 'react'

// ---| core |---
import { cn } from 'tools'

// ---| pages |---
// ---| screens |---
// ---| components |---
import Layout, { LayoutProps } from 'components/layouts/Layout'

// ---| self |---
import css from './Header.module.scss'

export type HeaderProps = LayoutProps

/**
 * Component description.
 *
 * How to use
 * @example
 * <Header />
 */
export function Header(props: HeaderProps) {
  const { children, className, ...otherProps } = props
  const _className = cn(css.Header, className)

  return <Layout className={_className} area='top' v='cr' aligns='center' {...otherProps}>{children}</Layout>
}

Header.displayName = 'Header'

export default Header
