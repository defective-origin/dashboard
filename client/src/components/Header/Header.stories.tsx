// eslint-disable-next-line no-restricted-imports
import { field, params } from '../../../.storybook/tool'
import type { Meta, StoryObj } from '@storybook/react'
import Header from './Header.component'

const meta: Meta<typeof Header> = {
  component: Header,
  title: 'Components/LAYOUT/MAGNETS/Header',
  tags: ['autodocs'],
  argTypes: {
    className: field.string(),
    children: field.reactNode(),
  },
}

export default meta

type Story = StoryObj<typeof Header>

export const Demo: Story = {
  parameters: params('Header'),
  args: {
    className: '',
    children: 'Demo',
  },
}
