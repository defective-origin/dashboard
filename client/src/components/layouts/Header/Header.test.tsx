// ---| tests |---
import { render } from 'tests'

// ---| self |---
import Header from './Header.component'
// import useHeader from './Header.hooks'
// import HeaderProvider, Header from './Header.context'

describe('[Header] component', () => {
  it('should render component', () => {
    const container = render(<Header />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
