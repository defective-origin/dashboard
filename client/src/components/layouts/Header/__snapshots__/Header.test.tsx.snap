// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[Header] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="block block--x item _Header_f4f7a7"
    style="align-items: center; grid-area: top;"
  />
</DocumentFragment>
`;
